independent-projects/resteasy-reactive/client/runtime/src/main/java/org/jboss/resteasy/reactive/client/handlers/ClientSendRequestHandler.java

```diff
@@ -358,7 +358,7 @@ public void handle(AsyncResult<Buffer> ar) {
                                         try {
                                             if (buffer.length() > 0) {
                                                 requestContext.setResponseEntityStream(
-                                                        new ByteBufInputStream(buffer.getByteBuf()));
+                                                        new ByteBufInputStream(buffer.getByteBuf(), true));
```

Comment:
closing an unreleaseable buffer won't have any effect AFAIK, but is fine to do it.
In case the buffer is not releaseable, instead.
The problem arise if the src buffer is used by the caller elsewhere, since we `slice` it without increasing the ref cnt, when the stream is closed, we release the src as well - which can be dangerous.
I think the problem here is the Vertx API which doesn't help to understand what to do with this...

Reply:
This codepath should not result in the buffer being reused anywhere else other than were we use it